tests/std/tests/GH_003570_allocate_at_least/test.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <deque>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <syncstream>
+#include <vector>
+
+using namespace std;
+
+bool allocate_at_least_called = false;
+
+bool consume_signal() {
+    bool ret                 = allocate_at_least_called;
+    allocate_at_least_called = false;
+    return ret;
+}

```

Comment:
We can more explicitly express the exchange semantics. (Note that `<utility>` should also be included.)
```suggestion
bool consume_signal() {
    static constinit bool allocate_at_least_called = false;
    return exchange(allocate_at_least_called, false);
}
```

Reply:
Doesn't work exactly as suggested since `set_signal` also needs access, but I've change it to a slightly more elaborate system utilizing `exchange`. Thanks :)